lib/internal/repl.js

```diff
@@ -55,15 +55,26 @@ function createRepl(env, opts, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
+  if (opts.terminal) {
     return setupHistory(repl, env.NODE_REPL_HISTORY,
                         env.NODE_REPL_HISTORY_FILE, cb);
   }
+
   repl._historyPrev = _replHistoryMessage;
   cb(null, repl);
 }
 
 function setupHistory(repl, historyPath, oldHistoryPath, ready) {
+  // Empty string disables persistent history.
+  if (historyPath === '') {
+    repl._historyPrev = _replHistoryMessage;
+    return ready(null, repl);
+  }
+
+  if (historyPath) {
```

Comment:
This condition is not necessary. If the control reaches this point, then `historyPath` is not an empty string.